### ISSUE TITLE
TT-RSS sync broken doesn't update all articles

### ISSUE THREAD

**Issue by mahdi1234** [martinrotter/rssguard#26]
#### How to reproduce the bug?

- wiped DB completely to avoid problem with migration from previous version
- setup new TT-RSS account and synced-in

Then did three updates in a row

- 1st one 43 unread
- 2nd one 72 unread
- 3rd one 74 unread

There were 172 unread articles at that time on server! Seems something totally broken in update, yes it's fast, but it doesn't work ...

Version: 3.3.1 (built on Linux/i686)
Revision: 45d20ba
Build date: 6/25/16 8:18 PM
Qt: 5.7.0 (compiled against 5.7.0)

Attached log and config (had to rename to config.txt as this great system doesn't allow ini suffix wtf?)

[log_rssguard.txt](https://github.com/martinrotter/rssguard/files/333231/log_rssguard.txt)

[config.txt](https://github.com/martinrotter/rssguard/files/333232/config.txt)

**Comment by mahdi1234** [comment]
Same problem when in-memory is not used.

**Comment by martinrotter** [comment]
Did you have checkbox "Force server-side update" checked?

**Comment by martinrotter** [comment]
And followup question: 

* 1st one 43 unread
* 2nd one 72 unread
* 3rd one 74 unread

You tried to update three times. When you tried for second and third time, were there any new messages for feeds which received messages the first time? Or when you updated, completely different feeds received new messages?

**Comment by mahdi1234** [comment]
No, I'm not using "Force server-side update" tt-rss server has its own cron, so I don't want client to update it.

I don't know whether those were completely different or additional news for feeds with some messages. I'm off now for two weeks so sorry don't have much time to look into this.

**Comment by martinrotter** [comment]
No problem.

**Comment by martinrotter** [comment]
Can you test latest available code c0824c3885db36cb2c9071c874090eb9b9fb3075 to check if the bug is solved?

**Comment by mahdi1234** [comment]
After short testing sync with tt-rss seems fine now.

Version: 3.3.2 (built on Linux/i686)
Revision: 4ef3e00
Build date: 6/29/16 10:31 PM
Qt: 5.7.0 (compiled against 5.7.0)

**Comment by martinrotter** [comment]
Thank you. I did TT-RSS testing by myself too. I also noticed this. Maybe it is just my TT-RSS instance fault, but sometimes TT-RSS just does not return "all" messages from given feed in its HTTP response, which sometimes result in not-matching numbers of unread messages (particularly of old messages).

Just saying that some problems might be actually caused by TT-RSS instance and not by RSS Guard itself.
